pkg/network/ebpf/c/protocols/postgres/decoding.h

```diff
@@ -107,12 +108,42 @@ static int __always_inline skip_string(pktbuf_t pkt, int message_len) {
     return SKIP_STRING_FAILED;
 }
 
+// Return a pointer to the postgres telemetry record in the corresponding map.
+static __always_inline void* get_pg_msg_counts_map(pktbuf_t pkt) {
+    const __u32 zero = 0;
+
+    pktbuf_map_lookup_option_t pg_telemetry_lookup_opt[] = {
+        [PKTBUF_SKB] = {
+            .map = &postgres_kernel_msg_count_plain,
+            .key = (void*)&zero,
+        },
+        [PKTBUF_TLS] = {
+            .map = &postgres_kernel_msg_count_tls,
+            .key = (void*)&zero,
+        },
+    };
+    return pktbuf_map_lookup(pkt, pg_telemetry_lookup_opt);
+}
+
+// update_pg_message_count_telemetry - increments the corresponding statistics counter
+//      according to the number of Postgres messages detected.
+static __always_inline void update_pg_message_count_telemetry(postgres_kernel_msg_count_t* pg_msg_counts, __u8 msg_count) {
+
+    __u8 bucket_idx = PG_KERNEL_MSG_COUNT_BUCKET_INDEX((__u8)msg_count);
+    bucket_idx = bucket_idx >= PG_KERNEL_MSG_COUNT_NUM_BUCKETS ? (PG_KERNEL_MSG_COUNT_NUM_BUCKETS - 1) : bucket_idx;
+    __sync_fetch_and_add(&pg_msg_counts->pg_messages_count_buckets[bucket_idx], 1);
+
+    return;
+}
```

Comment:
The current implementation maps `0 <= msg_count <5` into the first bucket, was that your original intention?

Reply:
yes, it was intention, 8 buckets of 5 counts in each